### PR TITLE
fix: generate correct changelogs when doing a PR

### PR DESCRIPTION
**Pull request by Nemikolh**
This PR fix the changelog generation so that it when we do `pnpm changelog` it generate the changelog for every packages, in particular the CLI. The previous release PR didn't had the changelog for the CLI because we are bumping the CLI at a later point.

### CODE REVIEW

File: scripts/changelog/Package.mjs [hunk 1]
```diff
@@ -0,0 +1,37 @@
+import fs from 'node:fs';
+import path from 'node:path';
+
+export class Package {
+  /**
+   * @param {PackageDefinition} pkg
```
Comment (AriPerkkio) [inline scripts/changelog/Package.mjs:6]: This should be `PackageConfiguration`?

```suggestion
   * @param {PackageConfiguration} pkg
```

To be able to reference that, let's move it into this file from `changelog.mjs` instead:

```js
/**
 * @typedef {{
 *  path: string;
 *  excluded?: true;
 *  sameAs?: string;
 * }} PackageConfiguration A package in the monorepo to generate changelog for.
 */

export class Package {
  /**
   * @param {PackageConfiguration} pkg
   */
```

Then use it in `changelog.mjs`:

```js
/**
 * @type {import('./changelog/Package.mjs').PackageConfiguration[]}
 */
const PACKAGES = [
  { path: './packages/astro' },
  ...
```

File: scripts/changelog/generate.mjs [hunk 2]
```diff
@@ -0,0 +1,70 @@
+import fs from 'node:fs';
+import addStream from 'add-stream';
+import chalk from 'chalk';
+import conventionalChangelog from 'conventional-changelog';
```
Comment (AriPerkkio) [inline scripts/changelog/generate.mjs:4]: FYI the package I mentioned earlier: https://github.com/antfu/changelogithub. It creates changelogs like https://github.com/unocss/unocss/releases/tag/v0.39.0.

**Review by AriPerkkio — APPROVED** [review]
Looks good, tested locally and seems to work nicely. 

<details>

```bash
$ pnpm --recursive --filter "@tutorialkit/*" exec pnpm version --no-git-tag-version --allow-same-version 0.0.1

v0.0.1
v0.0.1
v0.0.1
v0.0.1
v0.0.1

$ pnpm changelog

UPDATED packages/runtime/CHANGELOG.md 0.0.1
UPDATED packages/cli/CHANGELOG.md 0.0.1
UPDATED packages/astro/CHANGELOG.md 0.0.1
UPDATED packages/components/react/CHANGELOG.md 0.0.1
UPDATED packages/theme/CHANGELOG.md 0.0.1
UPDATED packages/types/CHANGELOG.md 0.0.1
UPDATED packages/create-tutorial/CHANGELOG.md 0.0.1-alpha.23
UPDATED CHANGELOG.md 0.0.1

$ git status

  modified:   CHANGELOG.md
  modified:   packages/astro/CHANGELOG.md
  modified:   packages/cli/CHANGELOG.md
  modified:   packages/components/react/CHANGELOG.md
  modified:   packages/runtime/CHANGELOG.md
  modified:   packages/theme/CHANGELOG.md
  modified:   packages/types/CHANGELOG.md
```


</details>

File: scripts/changelog/Package.mjs (hunk 1 above)
Comment (Nemikolh) [reply scripts/changelog/Package.mjs:6]: Oh this is nice! Thanks 😃

**Pull request merged**